internal/adapters/uidriver/socket.go

```
@@ -0,0 +1,256 @@
+	listener, err := net.ListenUnix("unix", address)
+	if err != nil {
+		return nil, fmt.Errorf("uidriver: listen on socket: %w", err)
+	}
+	// Never let net.UnixListener unlink the pathname behind our ownership
+	// check. A caller may replace the path after bind; cleanup must preserve
+	// that replacement rather than deleting it during Close.
+	listener.SetUnlinkOnClose(false)
+	boundInfo, infoErr := os.Lstat(path)
+	cleanup := func(cleanErr error) (*UnixEndpoint, error) {
+		_ = listener.Close()
+		removeOwnedSocket(path, boundInfo)
```

golangci-lint (2.13.2)

[error] Error return value is not checked

(errcheck)